tsvtools/ocrd_processors.py

```diff
@@ -0,0 +1,108 @@
+from json import loads
+from pathlib import Path
+from pkg_resources import resource_string
+from re import sub as re_sub
+
+import pandas as pd
+from PIL import Image
+
+from ocrd import Processor
+from ocrd_utils import getLogger, make_file_id, assert_file_grp_cardinality, MIMETYPE_PAGE
+from ocrd_models import OcrdExif
+from ocrd_models.constants import NAMESPACES as NS
+from ocrd_models.ocrd_page import TextEquivType, to_xml
+from ocrd_modelfactory import page_from_file
+
+from .cli import page2tsv
+
+OCRD_TOOL = loads(resource_string(__name__, 'ocrd-tool.json'))
+
+class OcrdNeatExportProcessor(Processor):
+
+    def __init__(self, *args, **kwargs):
+        kwargs['ocrd_tool'] = OCRD_TOOL['tools']['ocrd-neat-export']
+        kwargs['version'] = OCRD_TOOL['version']
+        super().__init__(*args, **kwargs)
+
+    def process(self):
+        """
+        Convert PAGE-XML to TSV loadable by the neat GT editor.
+        """
+        log = getLogger('ocrd_neat.export')
+        assert_file_grp_cardinality(self.input_file_grp, 1)
+        assert_file_grp_cardinality(self.output_file_grp, 1)
+        iiif_url_template = self.parameter['iiif_url_template']
+        noproxy = self.parameter['noproxy']
+
+        ppn_found = self.workspace.mets._tree.find('//mods:recordIdentifier[@source="gbv-ppn"]', NS)
+        print(ppn_found)
+        if ppn_found is not None:
+            ppn = ppn_found.text
+        else:
+            ppn = ''
+        for n, input_file in enumerate(self.input_files):
+            page_id = input_file.pageId or input_file.ID
+            log.info('Processing: %d / %s of %d', n, page_id, len(list(self.input_files)))
+            file_id = make_file_id(input_file, self.output_file_grp)
+            pcgts = page_from_file(self.workspace.download_file(input_file))
+            page = pcgts.get_Page()
+
+            iiif_url = iiif_url_template\
```

Comment:
Does that work universally? If so, we should probably write an IIIF image importer for OCR-D from scratch (instead of extending and using https://github.com/karkraeg/iiimets).

Reply:
To an extent. It is geared towards the ID conventions @StaatsbibliothekBerlin (`{{ PPN }}-{{ page_no }}`) but except from that, it is applicable to any IIIF URL scheme.